tests/test_table_transformations.py

```diff
@@ -239,6 +247,42 @@ def test_format_columns_with_the_same_name(self):
                 ['', '2'],
             ], columns=['Number', 'Number']))
 
+    def test_rename_columns(self):
+        src_df = pd.DataFrame(
+            [
+                ['P1', 'D1', 'BS1', 'BM1', 'TXT'],
+            ], columns=['patient.trial', 'PMC Diagnosis ID', 'PMC Biosource ID', 'PMC Biomaterial ID',
+                        '\\01. Patient\\Text\\'], index=[])
+
+        renamed_df = rename_columns(src_df, concept_path_to_name = {'\\01. Patient\\Text\\': 'Text'})
+
+        pdt.assert_frame_equal(renamed_df, pd.DataFrame(
+            [
+                ['P1', 'D1', 'BS1', 'BM1', 'TXT'],
+            ], columns=['Patient Id', 'Diagnosis Id', 'Biosource Id', 'Biomaterial Id',
+                    'Text'], index=[]))
+
+    def test_diagnossless_biosources(self):
+        self.test_data = [
+            ['BS1', 'D1', 'biosource_concept_1', '\\Patient\\Diagnosis\\Biosource\\Cell type\\', 'Cell type',
+             np.nan, 1, 'P1', 'Skin', 'TEST'],
+            ['BS2', None, 'biosource_concept_1', '\\Patient\\Diagnosis\\Biosource\\Cell type\\', 'Cell type',
+             np.nan, 2, 'P2', 'Liver', 'TEST']
```

Comment:
looks good! just out of curiosity, some tests use only `np.nan` (not only for numeric values), others `None` to represent missing values. This one has both. Is there some reason for that? Or are there some cases that better to use None over np.nan?

Reply:
I've replaced `nan`s with `None`s 😄